gramps_webapi/api/resources/citation.py

```diff
@@ -21,8 +21,15 @@ def object_extend(self, obj: Citation, args: Dict) -> Citation:
         """Extend citation attributes as needed."""
         if "extend" in args:
             db_handle = self.db_handle
-            obj.extended = get_extended_attributes(db_handle, obj)
-            obj.extended["source"] = get_source_by_handle(db_handle, obj.source_handle)
+            obj.extended = get_extended_attributes(db_handle, obj, args)
+            if (
+                "all" in args["extend"]
+                or "" in args["extend"]
```

Comment:
What is the rationale for the `"" in args["extend"]`? Is this in alias for `all`? It would trigger in cases like `extend=place,,father` (double comma).

Reply:
Yes that handles the ?extend case and treats it the same as extend=all

I had thought webargs DelimitedList should let me set the default if missing and that didn't seem to work and I wanted to try to preserve the other behaviour. A check of marshmallow docs doesn't show a DelimitedList so it is a webargs construct.

Good point on the double comma, I had not thought of that. I should change to `args["extend"] == [""]` or do you suggest a better way of handling?

Reply:
Did you check that `?extend` actually parses as `[""]`? And can this be relied on? It seems like a hack to me. `extend=all` is clean and unambiguos. And we don't have to worry about backwards compatibility :hand_over_mouth:

Reply:
I did and it does, but if you think it is cleaner I'll drop support for the plain jane ?extend case.

Reply:
I leave it up to you.

Reply:
I trust your instinct on dropping it, doing so now.